taskiq/cli/scheduler/run.py

```diff
@@ -142,32 +149,53 @@ async def run_scheduler_loop(scheduler: TaskiqScheduler) -> None:
     """
     loop = asyncio.get_event_loop()
     running_schedules = set()
+
+    interval_tasks_last_run: "Dict[str, datetime]" = {}
+    now = datetime.now()
+    next_minute = now.replace(second=0, microsecond=0)
+    next_second = now.replace(microsecond=0) + timedelta(seconds=1)
+
+    await asyncio.sleep((next_second - now).total_seconds())
+
     while True:
-        # We use this method to correctly sleep for one minute.
+        # We use this method to correctly sleep.
+        now = datetime.now()
         scheduled_tasks = await get_all_schedules(scheduler)
         for source, task_list in scheduled_tasks.items():
             for task in task_list:
-                try:
-                    task_delay = get_task_delay(task)
-                except ValueError:
-                    logger.warning(
-                        "Cannot parse cron: %s for task: %s, schedule_id: %s",
-                        task.cron,
-                        task.task_name,
-                        task.schedule_id,
-                    )
-                    continue
-                if task_delay is not None:
-                    send_task = loop.create_task(
-                        delayed_send(scheduler, source, task, task_delay),
-                    )
+                to_send = False
+                if task.time is not None and now >= task.time:
+                    to_send = True
+                elif task.interval is not None:
+                    last_call = interval_tasks_last_run.get(task.task_name)
+                    if (
+                        last_call is not None
+                        and round((now - last_call).total_seconds()) < task.interval
+                    ):
+                        to_send = False
```

Comment:
`to_send` is `False` by default. You can negate this if condition and only leave the second branch.